File: js/narniaArchenlanders.js
var nm1 = ["","","","b","c","d","g","l","m","n","r","s","sh","tr"];
var nm2 = ["a","o","u"];
var nm3 = ["b","br","d","dr","l","ll","lv","m","n","t","r","rr"];
var nm4 = ["e","i"];
var nm5 = ["","","","l","m","n","r"];

var nm6 = ["","","","","","b","f","h","l","m","n","r","t"];
var nm7 = ["a","i","a","i","a","i","e","e","o"];
var nm8 = ["d","g","h","l","m","n","r","t"];
var nm9 = ["l","m","n","r","t","v"];
var nm10 = ["ln","m","mn","n","nn","ph","rn","s","sh","t","th"];

var br = "";

function nameGen(type){
	$('#placeholder').css('textTransform', 'capitalize');
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");

	for(i = 0; i < 10; i++){
		if(tp === 1){
			rnd = Math.random() * nm6.length | 0;
			rnd2 = Math.random() * nm7.length | 0;
			rnd3 = Math.random() * nm10.length | 0;
			if(i < 4){
				while(rnd < 5){
					rnd = Math.random() * nm6.length | 0;
				}
				names = nm6[rnd] + nm7[rnd2] + nm10[rnd3];
			}else{
				rnd4 = Math.random() * nm8.length | 0;
				rnd5 = Math.random() * nm7.length | 0;
				if(i < 7){
					names = nm6[rnd] + nm7[rnd2] + nm8[rnd4] + nm7[rnd5] + nm10[rnd3];
				}else{
					rnd6 = Math.random() * nm9.length | 0;
					rnd7 = Math.random() * nm7.length | 0;
					names = nm6[rnd] + nm7[rnd2] + nm8[rnd4] + nm7[rnd5] + nm9[rnd6] + nm7[rnd7] + nm10[rnd3];
				}
			}
		}else{
			rnd = Math.random() * nm1.length | 0;
			rnd2 = Math.random() * nm2.length | 0;
			rnd3 = Math.random() * nm5.length | 0;
			if(i < 5){
				while(rnd < 3){
					rnd = Math.random() * nm1.length | 0;
				}
				if(rnd3 < 3){
					rnd3 = rnd3 + 3;
				}
				names = nm1[rnd] + nm2[rnd2] + nm5[rnd3];
			}else{
				rnd4 = Math.random() * nm3.length | 0;
				rnd5 = Math.random() * nm4.length | 0;
				names = nm1[rnd] + nm2[rnd2] + nm3[rnd4] + nm4[rnd5] + nm5[rnd3];
			}
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/narniaArchenlanders.py
__all__ = ['narniaArchenlanders']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm8', 'nm3', 'nm7', 'nm9', 'nm10', 'nm2', 'nm6'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['tp', 'result', 'type'])
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm6').get('length'))|Js(0.0)))
                var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm10').get('length'))|Js(0.0)))
                if (var.get('i')<Js(4.0)):
                    while (var.get('rnd')<Js(5.0)):
                        var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm6').get('length'))|Js(0.0)))
                    var.put('names', ((var.get('nm6').get(var.get('rnd'))+var.get('nm7').get(var.get('rnd2')))+var.get('nm10').get(var.get('rnd3'))))
                else:
                    var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm8').get('length'))|Js(0.0)))
                    var.put('rnd5', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                    if (var.get('i')<Js(7.0)):
                        var.put('names', ((((var.get('nm6').get(var.get('rnd'))+var.get('nm7').get(var.get('rnd2')))+var.get('nm8').get(var.get('rnd4')))+var.get('nm7').get(var.get('rnd5')))+var.get('nm10').get(var.get('rnd3'))))
                    else:
                        var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm9').get('length'))|Js(0.0)))
                        var.put('rnd7', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                        var.put('names', ((((((var.get('nm6').get(var.get('rnd'))+var.get('nm7').get(var.get('rnd2')))+var.get('nm8').get(var.get('rnd4')))+var.get('nm7').get(var.get('rnd5')))+var.get('nm9').get(var.get('rnd6')))+var.get('nm7').get(var.get('rnd7')))+var.get('nm10').get(var.get('rnd3'))))
            else:
                var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
                var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
                if (var.get('i')<Js(5.0)):
                    while (var.get('rnd')<Js(3.0)):
                        var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
                    if (var.get('rnd3')<Js(3.0)):
                        var.put('rnd3', (var.get('rnd3')+Js(3.0)))
                    var.put('names', ((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm5').get(var.get('rnd3'))))
                else:
                    var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm3').get('length'))|Js(0.0)))
                    var.put('rnd5', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                    var.put('names', ((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd5')))+var.get('nm5').get(var.get('rnd3'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js(''), Js(''), Js(''), Js('b'), Js('c'), Js('d'), Js('g'), Js('l'), Js('m'), Js('n'), Js('r'), Js('s'), Js('sh'), Js('tr')]))
var.put('nm2', Js([Js('a'), Js('o'), Js('u')]))
var.put('nm3', Js([Js('b'), Js('br'), Js('d'), Js('dr'), Js('l'), Js('ll'), Js('lv'), Js('m'), Js('n'), Js('t'), Js('r'), Js('rr')]))
var.put('nm4', Js([Js('e'), Js('i')]))
var.put('nm5', Js([Js(''), Js(''), Js(''), Js('l'), Js('m'), Js('n'), Js('r')]))
var.put('nm6', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('b'), Js('f'), Js('h'), Js('l'), Js('m'), Js('n'), Js('r'), Js('t')]))
var.put('nm7', Js([Js('a'), Js('i'), Js('a'), Js('i'), Js('a'), Js('i'), Js('e'), Js('e'), Js('o')]))
var.put('nm8', Js([Js('d'), Js('g'), Js('h'), Js('l'), Js('m'), Js('n'), Js('r'), Js('t')]))
var.put('nm9', Js([Js('l'), Js('m'), Js('n'), Js('r'), Js('t'), Js('v')]))
var.put('nm10', Js([Js('ln'), Js('m'), Js('mn'), Js('n'), Js('nn'), Js('ph'), Js('rn'), Js('s'), Js('sh'), Js('t'), Js('th')]))
pass
pass


# Add lib to the module scope
narniaArchenlanders = var.to_python()